RevenueCatUI/CustomerCenter/ViewModels/ManageSubscriptionsViewModel.swift

```diff
@@ -0,0 +1,185 @@
+
+    private var error: Error?
+
+    convenience init() {
+        self.init(purchasesProvider: ManageSubscriptionPurchases())
+    }
+
+    // @PublicForExternalTesting
+    init(purchasesProvider: ManageSubscriptionsPurchaseType) {
+        self.state = .notLoaded
+        self.purchasesProvider = purchasesProvider
+    }
+
+    // @PublicForExternalTesting
+    init(configuration: CustomerCenterConfigData,
+         subscriptionInformation: SubscriptionInformation) {
+        self.configuration = configuration
+        self.subscriptionInformation = subscriptionInformation
+        self.purchasesProvider = ManageSubscriptionPurchases()
+        state = .success
+    }
+
+    func loadScreen() async {
+        do {
+            try await loadSubscriptionInformation()
+            await loadCustomerCenterConfig()
+            self.state = .success
+        } catch {
+            self.state = .error(error)
+        }
+    }
+
+    func loadSubscriptionInformation() async throws {
+        let customerInfo = try await purchasesProvider.customerInfo()
+        guard let currentEntitlementDict = customerInfo.entitlements.active.first,
+              let subscribedProductID = customerInfo.activeSubscriptions.first,
+              let subscribedProduct = await purchasesProvider.products([subscribedProductID]).first else {
+            Logger.warning(Strings.could_not_find_subscription_information)
+            throw CustomerCenterError.couldNotFindSubscriptionInformation
+        }
+        let currentEntitlement = currentEntitlementDict.value
+
+        // swiftlint:disable:next todo
+        // TODO: support non-consumables
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateStyle = .medium
+        self.subscriptionInformation = SubscriptionInformation(
+            title: subscribedProduct.localizedTitle,
+            durationTitle: subscribedProduct.subscriptionPeriod?.durationTitle ?? "",
+            price: subscribedProduct.localizedPriceString,
+            nextRenewalString: currentEntitlement.expirationDate.map { dateFormatter.string(from: $0) } ?? nil,
+            willRenew: currentEntitlement.willRenew,
+            productIdentifier: subscribedProductID,
+            active: currentEntitlement.isActive
+        )
+    }
+
+    func loadCustomerCenterConfig() async {
+        self.configuration = CustomerCenterConfigTestData.customerCenterData
+    }
+
+    #if os(iOS) || targetEnvironment(macCatalyst)
+    func handleAction(for path: CustomerCenterConfigData.HelpPath) {
+        switch path.type {
+        case .missingPurchase:
+            self.showRestoreAlert = true
+        case .refundRequest:
+            Task {
+                guard let subscriptionInformation = self.subscriptionInformation else { return }
+                let productId = subscriptionInformation.productIdentifier
+                let status = try await purchasesProvider.beginRefundRequest(forProduct: productId)
```

Comment:
Currently, any errors thrown inside the task will disappear - they should be caught within the task. (If the method is made async and the task blocks removed, then the compiler will complain.)